### DISCUSSION TITLE
Get limited API key permissions details

### DISCUSSION THREAD

**Discussion by macraig** [meilisearch/product#660] [Feedback & Feature Proposal]
### Problem statement 
Currently, there is no straightforward way to determine the permissions associated with an API key other than trial and error across various endpoints. This approach is inefficient and lacks transparency for users using limited access keys.

The goal of this feature request is to enhance the API's transparency and manageability, particularly for those using limited access keys.

### Potential solution
Introduce a new endpoint that would return the permissions associated with the provided API key, detailing allowed actions and accessible resources. This would help developers when using limited access keys, providing transparency on potential action restrictions.

Another option is to enhance the current `GET /keys` [endpoint](https://www.meilisearch.com/docs/learn/security/master_api_keys#listing-api-keys) to accept limited keys and only return the information about the keys they can access.

**Comment by Bwilliamson55** [comment]
Yes please!
The /keys endpoint would make the most sense I think.

**Comment by curquiza** [comment]
Hello! Do we know what the other products do?
For example, with GitHub API key, when the administrator gives you the key, you have no way to know what the key is for. You have to trust the admin person.

**Comment by CreMindES** [comment]
Many other product implement a `can-i` endpoint, which is really useful in these cases.
